### PR TITLE
Add 'deleted' field to API.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Update API spec to return deleted flags for all the things that can be created or deleted.

## Test Plan

Update existing integration/migration tests to also check the new deleted field.

### CODE REVIEW

File: idb/postgres/postgres.go [hunk 1]
```diff
@@ -1638,6 +1638,7 @@ func (db *IndexerDb) yieldAccountsThread(req *getAccountsRequest) {
 		var rewardstotal uint64
 		var createdat sql.NullInt64
 		var closedat sql.NullInt64
+		var deleted sql.NullBool
```
[the remark this answers is not in the document]
Comment (winder) [reply idb/postgres/postgres.go:1641]: I mistakenly marked the field as `NOT NULL`, during the migration it needs to be null or else we could report incorrect values.

File: idb/dummy.go [hunk 2]
```diff
@@ -353,6 +353,7 @@ type AssetRow struct {
 	Error        error
 	CreatedRound *uint64
 	ClosedRound  *uint64
+	Deleted      *bool
```
[the remark this answers is not in the document]
Comment (winder) [reply idb/dummy.go:356]: unfortunately it needs to be true/false/unset :(

File: idb/postgres/postgres.go [hunk 3]
```diff
@@ -2549,7 +2598,8 @@ func (db *IndexerDb) yieldApplicationsThread(ctx context.Context, rows *sql.Rows
 		var paramsjson []byte
 		var created *uint64
 		var closed *uint64
-		err := rows.Scan(&index, &creator, &paramsjson, &created, &closed)
+		var deleted *bool
```
Comment (brianolson) [inline idb/postgres/postgres.go:2601]: why not sql.NullBool here? (and a couple other places; usage inconsistent)

**Review by brianolson — COMMENTED** [review]
inconsistent usage of Scan() into *bool or sql.NullBool
I don't like `*bool` as a type in Go; it's inefficient and I'd rather see an enum {Unknown,Deleted,Active} but we're short on time so I won't press it.

File: idb/postgres/postgres.go (hunk 3 above)
Comment (winder) [reply idb/postgres/postgres.go:2604]: I'm not sure actually, this is the pattern I used for CreatedAt/DeletedAt so I copied it here.

**Pull request merged**